### PR TITLE
create entities

### PR DESCRIPTION
**Pull request by KlaraAzimova**

### CODE REVIEW

File: src/main/java/kg/peaksoft/giftlistb6/entities/Category.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @Id
+    @SequenceGenerator(name = "category_seq", sequenceName = "category_seq", allocationSize = 1)
+    @GeneratedValue(generator = "category_seq", strategy = GenerationType.SEQUENCE)
+    private Long id;
+
+    private String name;
+
+    @ManyToOne
+    private SubCategory subCategory;
+    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.REFRESH, CascadeType.PERSIST, CascadeType.MERGE})
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Category.java:26]: Pls, write to the next line. Our code should be in one standard

File: src/main/java/kg/peaksoft/giftlistb6/entities/Charity.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private Long id;
+
+    private String name;
+
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "charity")
+    private List<Category> category;
+
+    @ManyToOne
+    private User reservoir;
+    @ManyToOne
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Charity.java]: Pls, write to the next line.

File: src/main/java/kg/peaksoft/giftlistb6/entities/Complaint.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@Table(name = "complaints")
+@Getter
+@Setter
+@NoArgsConstructor
+public class Complaint {
+    @Id
+    @SequenceGenerator(name = "complaint_seq", sequenceName = "complaint_seq", allocationSize = 1)
+    @GeneratedValue(generator = "complaint_seq", strategy = GenerationType.SEQUENCE)
+    private Long id;
+    @ManyToOne
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Complaint.java]: Pls, write to the next line.

File: src/main/java/kg/peaksoft/giftlistb6/entities/Category.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    private Charity charity;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Category.java:27]: How you will map enums?

File: src/main/java/kg/peaksoft/giftlistb6/entities/Complaint.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    private Wish wish;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Complaint.java:22]: How you will map enums?

File: src/main/java/kg/peaksoft/giftlistb6/entities/Complaint.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    @ManyToOne
+    private User complainer;
+
+    private Boolean seen;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Complaint.java]: isSeen

File: src/main/java/kg/peaksoft/giftlistb6/entities/Gift.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+public class Gift {
+    @Id
+    @SequenceGenerator(name = "gift_seq", sequenceName = "gift_seq", allocationSize = 1)
+    @GeneratedValue(generator = "gift_seq", strategy = GenerationType.SEQUENCE)
+    private Long id;
+
+    @ManyToOne
+    private User user;
+
+
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Gift.java]: Unnecessary line

File: src/main/java/kg/peaksoft/giftlistb6/entities/Holiday.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    private Long id;
+
+    private String name;
+
+    @Column(name = "date_of_holiday")
+    private LocalDate dateOfHoliday;
+
+    @Column(length = 10000)
+    private String image;
+    @ManyToOne
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Holiday.java:31]: Next line

File: src/main/java/kg/peaksoft/giftlistb6/entities/Notification.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    private LocalDate createdDate;
+
+    private Boolean seen;
+
+    @OneToOne
+    private User requestToFriend;
+
+    @OneToOne
+    private Wish wish;
+    @ManyToOne
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Notification.java]: Next line

File: src/main/java/kg/peaksoft/giftlistb6/entities/Notification.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+public class Notification {
+    @Id
+    @SequenceGenerator(name = "notification_seq", sequenceName = "notification_seq", allocationSize = 1)
+    @GeneratedValue(generator = "notification_seq", strategy = GenerationType.SEQUENCE)
+    private Long id;
+
+    @Column(name = "created_date")
+    private LocalDate createdDate;
+
+    private Boolean seen;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Notification.java]: isSeen

File: src/main/java/kg/peaksoft/giftlistb6/entities/User.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Column(unique = true)
+    private String email;
+
+    @Column(name = "is_block")
+    private Boolean isBlock;
+
+    private String photo;
+
+    private Role role;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/User.java:40]: How you will map enums?

File: src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    private Integer shoeSize;
+
+    @Column(name = "clothing_size")
+    private String clothingSize;
+
+    private String hobby;
+
+    private String important;
+
+    private String facebook;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java]: No need this field

File: src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+    private String instagram;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java]: No need

File: src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+    private String telegram;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java]: No need

File: src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+    private String vk;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java]: No need

File: src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+    private Long id;
+
+    @Column(name = "wish_name")
+    private String wishName;
+
+    @Column(name = "link_to_gift")
+    private String linkToGift;
+
+    @Column(name = "date_of_holiday")
+    private LocalDate dateOfHoliday;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java:31]: createdDate?

File: src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+    private String description;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java:34]: Add @Column(length = 10000)

File: src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+    @Column(name = "wish_status")
+    @Enumerated(EnumType.STRING)
+    private Status wishStatus;
+
+    @OneToOne
+    private User reservoir;
+
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "wish")
+    private List<Complaint> complaints;
+    @ManyToOne(cascade = {CascadeType.REFRESH
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java:49]: Next line

File: src/main/java/kg/peaksoft/giftlistb6/enums/Role.java [hunk 19]
```diff
@@ -0,0 +1,7 @@
+package kg.peaksoft.giftlistb6.enums;
+
+public enum Role {
+
+    ADMIN, USER;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/enums/Role.java]: Write USER to the next line pls

File: src/main/resources/application-klara.yml [hunk 20]
```diff
@@ -0,0 +1,11 @@
+spring:
```
Comment (i-Coderr) [inline src/main/resources/application-klara.yml]: Pls, don't push your config.yml file to the pull request. You can add it to the .gitignore file

**Review by i-Coderr — COMMENTED** [review]
Commented 11.10.22 at 3.20 pm

File: src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java (hunk 16 above)
Comment (KlaraAzimova) [reply src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java:31]: Нет, дата праздника.

File: src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Column(name = "shoe_size")
+    private int shoeSize;
+
+    @Column(name = "clothing_size")
+    private String clothingSize;
+
+    private String hobby;
+    private String important;
+    private String facebook;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java]: facebook, instagram, telegram, vk полялар кереги жок

File: src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+    private LocalDate dateOfBirth;
+
+    @Column(name = "shoe_size")
+    private int shoeSize;
+
+    @Column(name = "clothing_size")
+    private String clothingSize;
+
+    private String hobby;
+    private String important;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java:38]: next line

File: src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java (hunk 16 above)
Comment (i-Coderr) [reply src/main/java/kg/peaksoft/giftlistb6/entities/Wish.java:31]: ok

**Review by i-Coderr — COMMENTED** [review]
Enum'дар кайда кетти? Же башка enum болбойбу?

File: src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java (hunk 21 above)
Comment (KlaraAzimova) [reply src/main/java/kg/peaksoft/giftlistb6/entities/UserInfo.java]: ok

**Pull request merged**